src/CoreWCF.Primitives/src/CoreWCF/Channels/ReceiveContext.cs

```diff
@@ -8,7 +8,7 @@
 
 namespace CoreWCF.Channels
 {
-    internal abstract class ReceiveContext
+    public abstract class ReceiveContext
     {
```

Comment:
Now this is being made public, I think we should remove the sync versions of the methods. Otherwise everyone must override both.